### ISSUE TITLE
'needs help' message shown even after completion

### ISSUE THREAD

**Issue by amit3107** [learningequality/kolibri#5908]
### Error 

In the reports of a lesson- for few students, the report shows the students need help even if they have cleared the mastery criteria.

### Observed behavior

In the reports of a lesson- for few students, the report shows the students need help even if they have cleared the mastery criteria.
It shows completed in the tab though.
![Screenshot_20190828-101926](https://user-images.githubusercontent.com/38091824/63833854-98a67f00-c991-11e9-889f-067e88a730cd.png)
![IMG_20190828_100101696](https://user-images.githubusercontent.com/38091824/63833855-98a67f00-c991-11e9-8487-2ab66bffa094.jpg)
![IMG_20190828_095755100](https://user-images.githubusercontent.com/38091824/63833856-993f1580-c991-11e9-9e99-c2cb7aab348f.jpg)
![IMG_20190828_102005934](https://user-images.githubusercontent.com/38091824/63833858-993f1580-c991-11e9-8ad5-d3f4b17508ac.jpg)
![IMG_20190828_100042598](https://user-images.githubusercontent.com/38091824/63833859-99d7ac00-c991-11e9-9ab4-8b633301eae5.jpg)
![IMG_20190828_101954943](https://user-images.githubusercontent.com/38091824/63833861-9b08d900-c991-11e9-9102-0a3b7c450626.jpg)

### Expected behavior

The report must show completed for all the students who met the mastery criteria. It must not show needs help for students who met the mastery criteria.
It must show completed on tab also.

### User-facing consequences

Teachers get confused that some students still need help even if they have mastered the exercise.

### Errors and logs
[kolibri.txt](https://github.com/learningequality/kolibri/files/3549480/kolibri.txt)

![e1](https://user-images.githubusercontent.com/38091824/63832716-08673a80-c98f-11e9-8e86-9a7431be9848.png)
![e2](https://user-images.githubusercontent.com/38091824/63832719-08ffd100-c98f-11e9-9625-09784e134704.png)
![e3](https://user-images.githubusercontent.com/38091824/63832720-09986780-c98f-11e9-91a2-3aa42b6ba69d.png)
![e4](https://user-images.githubusercontent.com/38091824/63832722-0a30fe00-c98f-11e9-97eb-f755e5cd1586.png)

### Steps to reproduce

Conduct a testing environment for 15-20 users and ask the users to master the exercises and make mistakes in some questions. Someone must view the reports to see the status of students for whom, needs help is getting displayed.

### Context

Tell us about your environment, including:
 * Kolibri version- 0.12.8
 * Operating system ubuntu 14.04
 * Browser google chrome

**Comment by amit3107** [comment]
@rtibbles @Miyer03
@indirectlylit

**Comment by miyer03** [comment]
This seems to be a bit random. Sometimes when the kid has attained Mastery by answering consecutive 4 questions correct, it shows needs help, and at time, when he has not obtained 4 correct question in a row it shows completed. The Mastery is 4 questions correct.

![IMG_20190829_122747](https://user-images.githubusercontent.com/8583367/64059575-101c1e80-cbdd-11e9-9b3a-3107cf3f87af.jpg)
![IMG_20190829_122849](https://user-images.githubusercontent.com/8583367/64059576-10b4b500-cbdd-11e9-81f9-8c5dcb3ab81d.jpg)
![IMG_20190830_164021](https://user-images.githubusercontent.com/8583367/64059577-10b4b500-cbdd-11e9-85ea-1bf8a932974c.jpg)
![IMG_20190829_123326](https://user-images.githubusercontent.com/8583367/64059578-10b4b500-cbdd-11e9-9555-24d2ed25546c.jpg)
![IMG_20190829_123249](https://user-images.githubusercontent.com/8583367/64059579-114d4b80-cbdd-11e9-8cfd-cf5248e9b71d.jpg)

**Comment by jonboiser** [comment]
It's possible for this to happen in exercises if the Learner has mastered an exercise, but then made more wrong attempts afterwards:

1. Master the exercise -> exercise completed
1. Go to the next question
1. Provide wrong answers to that question -> exercise now "needs help"

@amit3107 	Do you think something like that might be happening for some of the learners?

**Comment by amit3107** [comment]
A child got a star for an exercise after completing 5 questions correctly. Then she made 2 wrong attempts in the same exercise. The star turned into needs help in the server but remained as a star in the tab.
What you are saying makes sense.

**Comment by jonboiser** [comment]
@amit3107 	Thanks for the update. I can see how this behavior can be confusing. 

Looking at some of the screen shots, I think I can see other students might have been doing something similar. For example, they will have 4 green questions but then have several red ones after, which might cause this to happen.

**Comment by amit3107** [comment]
> @amit3107 Thanks for the update. I can see how this behavior can be confusing.
> 
> Looking at some of the screen shots, I think I can see other students might have been doing something similar. For example, they will have 4 green questions but then have several red ones after, which might cause this to happen.

Yes absolutely, I also feel this is the trend, but seeing the needs help tag confuses the teacher. Suggestion- It can show need help if the student gets 4 or more incorrect.

**Comment by indirectlylit** [comment]
Thanks for the information. I've logged an internal note so that we can improve the user experience in the future. It's not on the roadmap yet, but we'll update this issue here when that happens.
